2010-11-04  Michael Tautschnig  <mt@debian.org>

	* setup-storage/Sizes.pm: Do not rely on bios_sectors_per_track information
		for computing the start location of first partition, always use 63 sectors
		gap. Thanks Mathieu Alorent for extensive testing.
Index: trunk/lib/setup-storage/Sizes.pm
===================================================================
--- trunk.orig/lib/setup-storage/Sizes.pm
+++ trunk/lib/setup-storage/Sizes.pm	
@@ -641,15 +641,11 @@
     my $next_start = 0;
 
     if ($FAI::configs{$config}{disklabel} eq "msdos") {
-      # on msdos disk labels, the first partitions starts at head #1
-      $next_start = $current_disk->{bios_sectors_per_track} *
-        $current_disk->{sector_size};
+      # on msdos disk labels, the first partitions starts at head #1; well,
+      # enforce a 63-sectors-per-track layout
+      $next_start = 63 * $current_disk->{sector_size};
       $min_req_total_space += $next_start;
 
-      # the MBR requires space, too
-      $min_req_total_space += $current_disk->{bios_sectors_per_track} *
-        $current_disk->{sector_size};
-
     } elsif ($FAI::configs{$config}{disklabel} eq "gpt") {
       # on GPT-EFI disk labels the first 34 and last 33 sectors must be left alone
       $next_start = 34 * $current_disk->{sector_size};
